services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -2180,6 +2206,13 @@ private Pair<Boolean, Version> addVersion(
 
           version.setViewConfig(store.getViewConfigs());
 
+          createChangeCaptureVersionTopicIfChangeCaptureEnabled(
```

Comment:
So instead of calling createchangeCaptureVersionTopicIfChangeCaptureEnabled, lets call something like constructViewResources (which then goes through the list of views and instantiates things, specifically wrapping what you've already coded.